sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/PartitionReceiver.cs

```diff
@@ -22,8 +27,261 @@ namespace Azure.Messaging.EventHubs.Primitives
+            PartitionId = partitionId;
+            InitialPosition = eventPosition;
+            RetryPolicy = options.RetryOptions.ToRetryPolicy();
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="PartitionReceiver"/> class.
+        /// </summary>
+        ///
+        /// <param name="consumerGroup">The name of the consumer group this client is associated with.  Events are read in the context of this group.</param>
+        /// <param name="partitionId">The identifier of the Event Hub partition from which events will be received.</param>
+        /// <param name="eventPosition">The position within the partition where the client should begin reading events.</param>
+        /// <param name="connection">The <see cref="EventHubConnection" /> connection to use for communication with the Event Hubs service.</param>
+        /// <param name="options">A set of options to apply when configuring the client.</param>
+        ///
+        public PartitionReceiver(string consumerGroup,
+                                 string partitionId,
+                                 EventPosition eventPosition,
+                                 EventHubConnection connection,
+                                 PartitionReceiverOptions options = default)
+        {
+            Argument.AssertNotNullOrEmpty(consumerGroup, nameof(consumerGroup));
+            Argument.AssertNotNullOrEmpty(partitionId, nameof(partitionId));
+            Argument.AssertNotNull(connection, nameof(connection));
+
+            options = options?.Clone() ?? new PartitionReceiverOptions();
+
+            OwnsConnection = false;
+            Connection = connection;
+            ConsumerGroup = consumerGroup;
+            PartitionId = partitionId;
+            InitialPosition = eventPosition;
+            RetryPolicy = options.RetryOptions.ToRetryPolicy();
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="PartitionReceiver"/> class.
+        /// </summary>
+        ///
+        protected PartitionReceiver()
+        {
+            OwnsConnection = false;
+        }
+
+        /// <summary>
+        ///   Closes the client.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        public virtual async Task CloseAsync(CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            IsClosed = true;
+
+            var clientHash = GetHashCode().ToString();
+            Logger.ClientCloseStart(typeof(PartitionReceiver), EventHubName, clientHash);
+
+            try
+            {
+                if (OwnsConnection)
+                {
+                    await Connection.CloseAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
I believe this should be `CancellationToken.None` here; we're already deep into shutting down and shouldn't abort here for an owned connection.